docs/trace/links-based-sampler/README.md

```diff
@@ -0,0 +1,173 @@
+about a system. Ideally, it would be desirable to sample all linked traces together.
+
+As one possible way to address this, this example shows how we can increase the
+likelihood of having complete traces across linked traces.
+
+## How does this sampling example work?
+
+We use a composite sampler that makes use of two samplers:
+
+1. A parent based sampler.
+2. A links based sampler.
+
+This composite sampler first delegates to the parent based sampler. If the
+parent based sampler decides to sample, then the composite sampler decides
+to sample. However, if the parent based sampler decides to drop, the composite
+sampler delegates to the links based sampler. The links based sampler decides
+to sample if the activity has any linked activities and if at least ONE of those
+linked activities is sampled.
+
+The links based sampler is not a probabilistic sampler. It is a biased sampler
+that decides to sample an activity if any of the linked contexts are sampled.
+
+## When should you consider such an option?  What are the tradeoffs?
+
+This may be a good option to consider if you want to get more complete traces
+across linked traces. However, there are a few tradeoffs to consider:
+
+- **Not guaranteed to give consistent sampling in all situations**: This
+approach doesn't guarantee that you will get complete traces across linked
+traces in all situations.
+
+Let's look at a couple of cases using the same producer-consumer example
+scenario. Let's say we have a producer activity (say with ID S1 in Trace T1) that
+produces a message and a consumer activity (say with ID S2 in Trace T2)  that
+consumes the message.
+
+Now, let's say that the producing activity S1 in trace T1 is sampled, say using the
+decision of a parent based sampler. Now, let's say that the activity S2 in trace
+T2 is not sampled based on the parent based sampler decision for T2. However,
+since this activity S2 in T2 is linked to the producing activity (S1 in T1) that
+is sampled, this mechanism ensures that the consuming activity (S2 in T2) will
+also get sampled.
+
+Alternatively, let's consider what happens if the producing activity S1 in
+trace T1 is not sampled, say using the decision of a parent based sampler.
+Now, let's say that the consuming activity S2 in trace T2 is sampled, based
+on the decision of a parent based sampler. In this case, we can see that
+activity S2 in trace T2 is sampled even though activity S1 in trace T1 is not
+sampled. This is an example of a situation where this approach is not helpful.
+
+Another example of a situation where you would get a partial trace is if the
+consuming activity S2 in trace T2 is not the root activity in trace T2. In this
+case, let's say there's a different activity S3 in trace T2 that is the root
+activity. Let's say that the sampling decision was activity S3 was to drop it.
+Now, since S2 in trace T2 links to S1 in trace T1, with this approach S2 will
```

Comment:
```suggestion
Now, since S2 in trace T2 links to S1 in trace T1, with this approach S2 will still 
```

Reply:
Updated the text to elaborate on which activity will be included and which one will not be included.